keras/engine/training.py

```diff
@@ -3295,6 +3295,11 @@ def get_layer(self, name=None, index=None):
             "Provide either a layer name or layer index at `get_layer`."
         )
 
+    def _return_output_dtype(self):
+        """this method was implemented in order to
+        fix a small bug in tf.keras.layer.Layer.compute_output_signature"""
+        return self.get_layer(index=-1)._compute_dtype
```

Comment:
This is not reliable. There is no guarantee that the "last layer" in the list of layers is actually the output layer.

You could make this property work for `Functional` and `Sequential` but not in the general case.